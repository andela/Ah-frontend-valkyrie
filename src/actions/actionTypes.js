const actionTypes = {
  SIGNUP: "SIGNUP",
  LOGIN: "LOGIN",
  SIGNUPFAIL: "SIGNUPFAIL",
  SIGNUPSUCCESS: "SIGNUPSUCCESS",
  PASSWORD_RESET_SUCCESS: "PASSWORD_RESET_SUCCESS",
  PASSWORD_RESET_FAILED: "PASSWORD_RESET_FAILED",
};

export default actionTypes;
