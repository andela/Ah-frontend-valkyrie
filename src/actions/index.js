import { loginSuccess } from './loginActions';

const logout = () => ( dispatch ) => {
  localStorage.clear();
  dispatch( loginSuccess( {} ) );
};

export default logout;
