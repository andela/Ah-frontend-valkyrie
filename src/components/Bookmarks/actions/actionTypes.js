const BookmarkActionTypes = {
    ADD_BOOKMARK: 'ADD_BOOKMARK',
    ADD_BOOKMARK_ERROR: 'ADD_BOOKMARK_ERROR',
    FETCH_BOOKMARKS: 'FETCH_BOOKMARKS',
    FETCH_BOOKMARKS_ERROR: 'FETCH_BOOKMARKS_ERROR',
    DELETE_BOOKMARK: 'DELETE_BOOKMARK',
    DELETE_BOOKMARK_ERROR: 'DELETE_BOOKMARK_ERROR'
};

export default BookmarkActionTypes;
