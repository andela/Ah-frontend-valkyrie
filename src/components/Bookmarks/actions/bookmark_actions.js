import BookmarkActionTypes from './actionTypes';
import axios from 'axios';

const token = window.localStorage.getItem('auth_token');
const setConfig = method => {
    const config = {
        method: method,
        headers: {
            'Content-Type': 'application/json',
            Authorization: `Bearer ${token}`,
        },
        mode : "no-cors",
        cache: 'no-cache',
    }
    return config;
};

const setUrl = slug => {
    const url =  `${ process.env.HOST }/articles/${slug}/bookmark`;
    return url;
}

export const bookmarkArticle = bookmark =>(
    {
        type: BookmarkActionTypes.ADD_BOOKMARK,
        payload: bookmark,
    }
);

export const bookmarkArticleError = error =>(
    {
        type: BookmarkActionTypes.ADD_BOOKMARK_ERROR,
        payload: error,
    }
);

export const fetchBookmarks = bookmarks => (
    {
        type: BookmarkActionTypes.FETCH_BOOKMARKS,
        payload: bookmarks
    }
);

export const fetchBookmarksError = error => (
    {
        type: BookmarkActionTypes.FETCH_BOOKMARKS_ERROR,
        payload: error
    }
)

export const deleteBookmark = message =>(
    {
        type: BookmarkActionTypes.DELETE_BOOKMARK,
        payload: message,
    }
);

export const deleteBookmarkError = error => (
    {
        type: BookmarkActionTypes.DELETE_BOOKMARK_ERROR,
        payload: error,
    }
);

export const bookmarkAction = slug => ( dispatch ) => {
    const url = setUrl(slug)
    return axios(url, setConfig('POST'))
      .then(response => {
        dispatch(bookmarkArticle(response.data));
      }
      )
      .catch(error => {
        dispatch(bookmarkArticleError(error.response)); 
        }  
      )
};

export const getBookmarks = article_slug => ( dispatch ) => {
    const get_url = `${ process.env.HOST }/articles/${article_slug}/bookmark`
    return axios({
        get_url,
        method: "GET",
        headers: {
          "Content-Type": "application/json",
          Authorization: `Bearer ${token}`,
        },
        mode: "no-cors",
        cache: "no-cache",
      },)
      .then(response => {
        dispatch(fetchBookmarks(response.data));
      }
      )
      .catch(error => {
        dispatch(fetchBookmarksError(error.response)); 
        }  
      )
}

export const unBookmark = (slug,bookmark_id) => ( dispatch ) => {
    const delete_url = `${ process.env.HOST }/articles/${ slug }/bookmark/${ bookmark_id }`
    return axios(delete_url, setConfig('DELETE') )
      .then(response => {
        dispatch(deleteBookmark(response.data));
      }
      )
      .catch(error => {
        dispatch(deleteBookmarkError(error.response));   
        }  
      )
}
