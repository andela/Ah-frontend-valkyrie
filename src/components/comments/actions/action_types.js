const commentsActionTypes = {
  FETCH_COMMENTS: "FETCH_COMMENTS",
  FETCH_COMMENTS_ERROR: "FETCH_COMMENTS_ERROR",
  ADD_COMMENT: "ADD_COMMENT",
  LIKE_COMMENT: "LIKE_COMMENT",
  LIKE_COMMENT_ERROR: "LIKE_COMMENT_ERROR",
  ADD_COMMENT_ERROR: "ADD_COMMENT_ERROR",
  DELETE_COMMENT: "DELETE COMMENT",
  DELETE_COMMENT_ERROR: "DELETE_COMMENT_ERROR",
  EDIT_COMMENT_SUCCESS: "EDIT_COMMENT_SUCCESS",
  EDIT_COMMENT_ERROR: "EDIT_COMMENT_ERROR",

};

export default commentsActionTypes;
