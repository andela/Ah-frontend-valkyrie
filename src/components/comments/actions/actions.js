import commentsActionTypes from './action_types';
import * as api from '../../utils/http';

const commentsUrl = 'articles/';
// synchronous actions
export const fetchCommentsSuccess = comments => (
  {
    type: commentsActionTypes.FETCH_COMMENTS,
    payload: comments,
  }
);

export const fetchCommentsFailure = error => (
  {
    type: commentsActionTypes.FETCH_COMMENTS_ERROR,
    payload: error,
  }
);

export const addCommentsSuccess = comment => (
  {
    type: commentsActionTypes.ADD_COMMENT,
    payload: comment,
  }
);

export const addCommentsFailure = error => (
  {
    type: commentsActionTypes.ADD_COMMENT_ERROR,
    payload: error,
  }
);

export const deleteCommentSuccess = message => (
  {
    type: commentsActionTypes.DELETE_COMMENT,
    payload: message,
  }
);

export const deleteCommentFailure = error => (
  {
    type: commentsActionTypes.DELETE_COMMENT_ERROR,
    payload: error,
  }
);

export const editCommentSuccess = editedComment => (
  {
    type: commentsActionTypes.EDIT_COMMENT_SUCCESS,
    payload: editedComment,
  }
);

export const editCommentFailure = error => (
  {
    type: commentsActionTypes.EDIT_COMMENT_ERROR,
    payload: error,
  }
);
// asynchronous actions with AJAX API requests
export const fetchComments = slug => ( dispatch ) => {
  return api.getResource( `${ commentsUrl }${ slug }/comments/` )
    .then( ( response ) => {
      dispatch( fetchCommentsSuccess( response.data ) );
    } )
    .catch( ( err ) => {
      dispatch( fetchCommentsFailure( err.response ) );
    } );
};

export const addComment = ( slug, comment ) => ( dispatch ) => {
   return api.createResource( `${ commentsUrl }${ slug }/comments/`, comment )
    .then( ( response ) => {
      dispatch( addCommentsSuccess( response.data ) );
    } )
    .catch( ( err ) => {
      dispatch( addCommentsFailure( err.response ) );
    } );
};

export const deleteComment = ( slug, id ) => ( dispatch ) => {
   return api.removeResource( `${ commentsUrl }${ slug }/comments/${ id }` )
    .then( ( ) => {
      const message = `comment ${ id } has been successfully deleted`;
      dispatch( deleteCommentSuccess( { message, id } ) );
    } )
    .catch( ( err ) => {
      dispatch( deleteCommentFailure( err.response.data.detail ) );
    } );
};

export const editComment = ( slug, id, commentToEdit ) => ( dispatch ) => {
   return api.updateResource( `${ commentsUrl }${ slug }/comments/${ id }`, commentToEdit )
    .then( ( response ) => {
      dispatch( editCommentSuccess( { response, commentToEdit, id } ) );
    } )
    .catch( ( err ) => {
      dispatch( editCommentFailure( err.response.data.errors.body ) );
    } );
};