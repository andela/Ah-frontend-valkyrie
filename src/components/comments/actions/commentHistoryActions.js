import axios from 'axios';

export const getLatestHistory