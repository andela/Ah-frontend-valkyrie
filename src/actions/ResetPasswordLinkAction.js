import axios from "axios";
import actionTypes from "./actionTypes";

export const passwordResetSuccess = message => (
  {
    type: actionTypes.PASSWORD_RESET_SUCCESS,
    payload: message,
  }
);

export const passwordResetFail = error => (
  {
    type: actionTypes.PASSWORD_RESET_FAILED,
    payload: error,
  }
);

export const resetPasswordAction = email => (dispatch) => {
  const url = `${process.env.HOST}/users/reset_password`;
  return axios({
    url,
    method: "POST",
    data: { email },
    headers: {
      "Content-Type": "application/json",
    },
    mode: "no-cors",
    cache: "no-cache",
  },
  )
    .then((response) => {
      dispatch(passwordResetSuccess(response.data.user.message));
    })
    .catch((error) => {
      dispatch(passwordResetFail(error.response.data.user.detail));
    },
    );
};
