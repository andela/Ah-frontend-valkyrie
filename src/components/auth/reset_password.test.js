import React from "react";
import expect from "expect";
import { mount } from "enzyme";
import { Provider } from "react-redux";
import ResetPassword from "./reset_password";
import { resetPasswordAction } from "../../actions/ResetPasswordLinkAction";

const storeFake = (state, action) => ({
  subscribe: jest.fn(),
  default: jest.fn(),
  getState: () => state,
  dispatch: () => action,
  resetPasswordAction: jest.fn(),

});
const props = {
  resetPasswordAction: jest.fn(),
};

const store = storeFake({
  passwordResetReducer:
    {
      user: {},
      message: "",
    },
});

describe("reset password functionality", () => {
  const wrapper = mount(
    <Provider store={store}>
      <ResetPassword {...props} />
    </Provider>);
  it("matches the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("handles state on submit", () => {
    const url = window.location;
    const arr = url.toString().split("/");
    const form = wrapper.find("#resetPasswordForm");
    form.simulate("submit");
    expect(window.location.href).toEqual(`${arr[0]}//${arr[2]}/`);
  });

  it("handles state on change", () => {
    const event = { target: { name: "testname", value: "testvalue" } };
    const textField = wrapper.find(".form-control").first();
    textField.simulate("change", event);
    expect(textField.instance().value).toEqual("testvalue");
  });
});
