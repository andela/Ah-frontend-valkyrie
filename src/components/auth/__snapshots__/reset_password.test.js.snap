// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reset password functionality matches the snapshot 1`] = `ReactWrapper {}`;
